app/src/main/kotlin/at/bitfire/davdroid/ui/account/AccountScreenModel.kt

```diff
@@ -62,26 +61,32 @@ class AccountScreenModel @AssistedInject constructor(
         fun create(account: Account): AccountScreenModel
     }
 
+    private val _showOnlyPersonal = MutableStateFlow(false)
+    val showOnlyPersonal = _showOnlyPersonal.asStateFlow()
+
+    private var _showOnlyPersonalLocked = MutableStateFlow(false)
+    val showOnlyPersonalLocked = _showOnlyPersonalLocked.asStateFlow()
+
+    /**
+     * Only acquire account settings on a worker thread!
+     */
+    private val accountSettings by lazy { accountSettingsFactory.create(account) }
+
+    private suspend fun reload() = withContext(Dispatchers.Default) {
+        _showOnlyPersonal.value = accountSettings.getShowOnlyPersonal()
+        _showOnlyPersonalLocked.value = accountSettings.getShowOnlyPersonalLocked()
+    }
```

Comment:
`reload` is not called when the model is created:

1. Open DAVx5 account.
2. Check _show only personal_. Only personal collections are shown.
3. Navigate back to Accounts overview.
4. Open DAVx5 account again.
5. _Show only personal_ is unchecked again (because the setting is not loaded).

Also I don't know whether the `reload` should be at this location. Maybe better

~~~
private val accountSettings
val invalidAccount …

private val _showOnlyPersonal
val showOnlyPersonal

private var _showOnlyPersonalLocked
val showOnlyPersonalLocked

fun reloadShowOnlyPersonal       // instead of generic reload
init { reloadShowOnlyPersonal() } 

fun setShowOnlyPersonal
~~~

Will become better when we have AccountSettings as flows (when we have them in the DB) ;)

Reply:
Right. I would still put the init at the start of the class though.